compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/Parser.java

```diff
@@ -75,13 +80,24 @@ public Parser(CompilerContext context) {
         this.pkgCache = PackageCache.getInstance(context);
     }
 
-    public BLangPackage parse(PackageSource pkgSource) {
+    public BLangPackage parse(PackageSource pkgSource, Path sourceRootPath) {
         PackageID pkgId = pkgSource.getPackageId();
         BLangPackage pkgNode = (BLangPackage) TreeBuilder.createPackageNode();
         this.pkgCache.put(pkgId, pkgNode);
-
-        pkgSource.getPackageSourceEntries()
-                .forEach(e -> pkgNode.addCompilationUnit(generateCompilationUnit(e, pkgId)));
+        for (CompilerInput sourceInput: pkgSource.getPackageSourceEntries()) {
+            if (ProjectDirs.isTestSource(((FileSystemSourceInput) sourceInput).getPath(), sourceRootPath,
+                                         pkgId.getName().value)) {
+                if (!pkgNode.containsTestablePkg()) {
```

Comment:
Add a comment on why this check is needed.

Reply:
Added a comment with 4e9c501